src/lib.rs

```diff
@@ -43,7 +43,79 @@ pub trait Node {
   fn hash(&self) -> &[u8];
 }
 
-/// Main constructor. Takes an instance of `HashMethods`.
+/// A stream that generates a merkle tree based on the incoming data.
+///
+/// ## Example
+/// ```rust
+/// use merkle_tree_stream::{DefaultNode, HashMethods, MerkleTreeStream, Node, PartialNode};
+/// use std::rc::Rc;
+/// use std::vec::Vec;
+///
+/// struct SimpleXorHashMethods;
```

Comment:
This is pretty clever! 

Perhaps a slightly better name would be to call this ` XorHashMethod` without the `Simple` prefix. If you're new to hashing algorithms this might not seem simple at all. Small nit, but might help make things a bit more friendly. Thanks!